### ISSUE TITLE
Install Docker gives 500 error

### ISSUE THREAD

**Issue by RattyDAVE** [jupyterhub/jupyterhub#2843]
When trying a standard install using

```
docker run -d -p 8000:8000 --name jupyterhub jupyterhub/jupyterhub jupyterhub
```

I get 500 error when going to `http://DOCKERIP:8000` and no login page.

I then removed the container and ran without the -d

```
docker run -p 8000:8000 --name jupyterhub jupyterhub/jupyterhub jupyterhub
```

And get the following when I Hit the login page.

```
# docker run -p 8000:8000 --name jupyterhub jupyterhub/jupyterhub jupyterhub
[I 2019-12-06 16:07:46.804 JupyterHub app:2232] Running JupyterHub version 1.0.1dev
[I 2019-12-06 16:07:46.810 JupyterHub app:2263] Using Authenticator: jupyterhub.auth.PAMAuthenticator-1.0.1dev
[I 2019-12-06 16:07:46.811 JupyterHub app:2263] Using Spawner: jupyterhub.spawner.LocalProcessSpawner-1.0.1dev
[I 2019-12-06 16:07:46.811 JupyterHub app:2263] Using Proxy: jupyterhub.proxy.ConfigurableHTTPProxy-1.0.1dev
[I 2019-12-06 16:07:46.832 JupyterHub app:1394] Writing cookie_secret to /srv/jupyterhub/jupyterhub_cookie_secret
[I 2019-12-06 16:07:46.876 alembic.runtime.migration migration:154] Context impl SQLiteImpl.
[I 2019-12-06 16:07:46.877 alembic.runtime.migration migration:161] Will assume non-transactional DDL.
[I 2019-12-06 16:07:46.940 alembic.runtime.migration migration:513] Running stamp_revision  -> 4dc2d5a8c53c
[I 2019-12-06 16:07:47.189 JupyterHub proxy:461] Generating new CONFIGPROXY_AUTH_TOKEN
[W 2019-12-06 16:07:47.192 JupyterHub app:1624] No admin users, admin interface will be unavailable.
[W 2019-12-06 16:07:47.193 JupyterHub app:1626] Add any administrative users to `c.Authenticator.admin_users` in config.
[I 2019-12-06 16:07:47.193 JupyterHub app:1655] Not using whitelist. Any authenticated user will be allowed.
[I 2019-12-06 16:07:47.279 JupyterHub app:2303] Initialized 0 spawners in 0.003 seconds
[W 2019-12-06 16:07:47.287 JupyterHub proxy:643] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
[I 2019-12-06 16:07:47.290 JupyterHub proxy:646] Starting proxy @ http://:8000
16:07:48.374 [ConfigProxy] info: Proxying http://*:8000 to (no default)
16:07:48.381 [ConfigProxy] info: Proxy API at http://127.0.0.1:8001/api/routes
16:07:48.517 [ConfigProxy] info: 200 GET /api/routes 
[I 2019-12-06 16:07:48.519 JupyterHub app:2552] Hub API listening on http://127.0.0.1:8081/hub/
16:07:48.523 [ConfigProxy] info: 200 GET /api/routes 
[I 2019-12-06 16:07:48.524 JupyterHub proxy:320] Checking routes
[I 2019-12-06 16:07:48.525 JupyterHub proxy:400] Adding default route for Hub: / => http://127.0.0.1:8081
16:07:48.532 [ConfigProxy] info: Adding route / -> http://127.0.0.1:8081
16:07:48.534 [ConfigProxy] info: Route added / -> http://127.0.0.1:8081
16:07:48.536 [ConfigProxy] info: 201 POST /api/routes/ 
[I 2019-12-06 16:07:48.537 JupyterHub app:2627] JupyterHub is now running at http://:8000
```

Then when hitting the login page

```
[I 2019-12-06 16:22:16.203 JupyterHub log:174] 302 GET / -> /hub/ (@::ffff:192.168.11.143) 3.48ms
[I 2019-12-06 16:22:16.264 JupyterHub log:174] 302 GET /hub/ -> /hub/login (@::ffff:192.168.11.143) 2.26ms
[E 2019-12-06 16:22:16.402 JupyterHub web:1788] Uncaught exception GET /hub/login (::ffff:192.168.11.143)
    HTTPServerRequest(protocol='http', host='192.168.11.12:8000', method='GET', uri='/hub/login', version='HTTP/1.1', remote_ip='::ffff:192.168.11.143')
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/tornado/web.py", line 1699, in _execute
        result = await result
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/login.py", line 135, in get
        self.finish(self._render(username=username))
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/login.py", line 102, in _render
        {'next': self.get_argument('next', '')},
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 1119, in render_template
        template = self.get_template(name)
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 1113, in get_template
        return self.settings['jinja2_env'].get_template(name)
      File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 830, in get_template
        return self._load_template(name, self.make_globals(globals))
      File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 804, in _load_template
        template = self.loader.load(self, name, globals)
      File "/usr/local/lib/python3.6/dist-packages/jinja2/loaders.py", line 408, in load
        raise TemplateNotFound(name)
    jinja2.exceptions.TemplateNotFound: login.html
    
[E 2019-12-06 16:22:16.411 JupyterHub web:1216] Uncaught exception in write_error
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/tornado/web.py", line 1214, in send_error
        self.write_error(status_code, **kwargs)
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 1203, in write_error
        html = self.render_template('error.html', **ns)
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 1119, in render_template
        template = self.get_template(name)
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 1113, in get_template
        return self.settings['jinja2_env'].get_template(name)
      File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 830, in get_template
        return self._load_template(name, self.make_globals(globals))
      File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 804, in _load_template
        template = self.loader.load(self, name, globals)
      File "/usr/local/lib/python3.6/dist-packages/jinja2/loaders.py", line 408, in load
        raise TemplateNotFound(name)
    jinja2.exceptions.TemplateNotFound: error.html
    
[E 2019-12-06 16:22:16.414 JupyterHub log:166] {
      "X-Forwarded-Host": "192.168.11.12:8000",
      "X-Forwarded-Proto": "http",
      "X-Forwarded-Port": "8000",
      "X-Forwarded-For": "::ffff:192.168.11.143",
      "Cookie": "portainer.datatable_text_filter_home_endpoints=[secret]; portainer.datatable_settings_containers=[secret]; portainer.datatable_pagination_containers=[secret]; portainer.datatable_settings_images=[secret]; portainer.datatable_pagination_images=[secret]; portainer.datatable_settings_container-networks=[secret]; sickchill_user=[secret]; portainer.LOGIN_STATE_UUID=[secret]",
      "Accept-Language": "en-GB,en;q=0.9",
      "Accept-Encoding": "gzip, deflate",
      "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
      "User-Agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.97 Safari/537.36",
      "Upgrade-Insecure-Requests": "1",
      "Dnt": "1",
      "Connection": "close",
      "Host": "192.168.11.12:8000"
    }
[E 2019-12-06 16:22:16.415 JupyterHub log:174] 500 GET /hub/login (@::ffff:192.168.11.143) 14.10ms
```
All IP address are internal and not public. So no point in blanking out.

Any assistance would be good!

Thanks

DAVE

**Comment by ryogesh** [comment]
Looks like a networking issue, can you refer
https://jupyterhub.readthedocs.io/en/stable/getting-started/networking-basics.html#configure-the-hub-if-the-proxy-or-spawners-are-remote-or-isolated

**Comment by realjago** [comment]
Some here, it's not a networking issue.
Use jupyterhub/jupyterhub:1.0.0 and it works.
The master/latest/1.0 is broken, there is also no conda installed

**Comment by consideRatio** [comment]
It looks like it search for templates to render into html that isnt found, and that an assumption is broken about such template availability.

**Comment by jtpio** [comment]
This is indeed happening with `jupyterhub/jupyterhub:1.1`.

But it seems to be fixed in `jupyterhub/jupyterhub:1.2`.
